### DISCUSSION TITLE
특정 채팅방에 메시지 보내기 (v4)

### DISCUSSION THREAD

**Discussion by jaeeewon** [storycraft/node-kakao#516] [Q&A] [answered]
var channel = client.channelManager.get(id);
channel.sendText(chat.text);

<<-- v3 에서는 위와 같이 해도 된다고 합니다.

 v4 에서는 어떻게 해야 할지 도움 부탁드립니다..

여기 저기 찾아 보아도 node-kakao 관련 내용을 얻기가 어렵네요 ...

**Comment by ghost** [comment, marked as answer]
```typescript
let otherClient:any = client.channelList.get(roomID); //roomID must be Long Type
otherClient.sendChat("TEST");
```

이런식으로 하면 될껍니다.

**Reply by jaeeewon** [reply]
> 감사합니다~!!

**Reply by wiresharkkkk** [reply]
any부분에 에러표시가 나는데 
any를 지우고
let otherClient = client.channelList.get(채널아이디);
otherClient.sendChat("1111");
이렇게 쓰면 작동이 안되는데 어떻게 해줘야할까용..?

**Reply by secneuqes** [reply]
> any부분에 에러표시가 나는데 
> any를 지우고
> let otherClient = client.channelList.get(채널아이디);
> otherClient.sendChat("1111");
> 이렇게 쓰면 작동이 안되는데 어떻게 해줘야할까용..?

any는 타입스크립트에서 타입을 지정해주는 것이라 삭제해주셔도 되고, sendChat 값은 작은 따옴표로 감싸고  방 아이디 값을 넣어주면 됩니다.('1037203749027392') <- 이런식으로요

**Reply by wiresharkkkk** [reply]
> > any부분에 에러표시가 나는데
> > any를 지우고
> > let otherClient = client.channelList.get(채널아이디);
> > otherClient.sendChat("1111");
> > 이렇게 쓰면 작동이 안되는데 어떻게 해줘야할까용..?
> 
> any는 타입스크립트에서 타입을 지정해주는 것이라 삭제해주셔도 되고, sendChat 값은 작은 따옴표로 감싸고 방 아이디 값을 넣어주면 됩니다.('1037203749027392') <- 이런식으로요

console.log(`${channel.channelId}`)
이렇게해서 나온 숫자대로
let otherClient = client.channelList.get('18325161476039326');
otherClient.sendChat('test');
이렇게 써보았는데 뭐가 잘못된건지 감이 안잡히네요 ㅠㅠ

**Reply by ghost** [reply]
> > > any부분에 에러표시가 나는데
> > > any를 지우고
> > > let otherClient = client.channelList.get(채널아이디);
> > > otherClient.sendChat("1111");
> > > 이렇게 쓰면 작동이 안되는데 어떻게 해줘야할까용..?
> > 
> > 
> > any는 타입스크립트에서 타입을 지정해주는 것이라 삭제해주셔도 되고, sendChat 값은 작은 따옴표로 감싸고 방 아이디 값을 넣어주면 됩니다.('1037203749027392') <- 이런식으로요
> 
> console.log(`${channel.channelId}`)
> 이렇게해서 나온 숫자대로
> let otherClient = client.channelList.get('18325161476039326');
> otherClient.sendChat('test');
> 이렇게 써보았는데 뭐가 잘못된건지 감이 안잡히네요 ㅠㅠ

get의 매게변수 타입은 Long입니다. new Long('18325161476039326') 또는 Long.fromString('18325161476039326') 을 통해 Long타입으로 타입캐스팅해주세요

**Reply by wiresharkkkk** [reply]
> > > > any부분에 에러표시가 나는데
> > > > any를 지우고
> > > > let otherClient = client.channelList.get(채널아이디);
> > > > otherClient.sendChat("1111");
> > > > 이렇게 쓰면 작동이 안되는데 어떻게 해줘야할까용..?
> > > 
> > > 
> > > any는 타입스크립트에서 타입을 지정해주는 것이라 삭제해주셔도 되고, sendChat 값은 작은 따옴표로 감싸고 방 아이디 값을 넣어주면 됩니다.('1037203749027392') <- 이런식으로요
> > 
> > 
> > console.log(`${channel.channelId}`)
> > 이렇게해서 나온 숫자대로
> > let otherClient = client.channelList.get('18325161476039326');
> > otherClient.sendChat('test');
> > 이렇게 써보았는데 뭐가 잘못된건지 감이 안잡히네요 ㅠㅠ
> 
> get의 매게변수 타입은 Long입니다. new Long('18325161476039326') 또는 Long.fromString('18325161476039326') 을 통해 Long타입으로 타입캐스팅해주세요

`
let otherClient = CLIENT.channelList.get(Long.fromString('18325161476039326'));
`
`
otherClient.sendChat('test');
`
채널id가 틀렸거나 채널에 입장되지 않은 문제는 아닐텐데 이상하네요..
